docs/syncing-forks.md

```diff
@@ -0,0 +1,149 @@
+helps maintaining commit history and makes pulling future changes easier.
+
+## One time setup
+
+### Pull forked repo to your local machine
+
+```console
+# could be MS runtime or another MS app
+$ git clone git@github.com:mobilestack-xyz/mobilestack-runtime.git
+```
+
+### Add upstream remote repository
+
+On the root of the forked repository
+
+```console
+# if pulling changes from Valora to MS run time
+$ git remote add upstream git@github.com:valora-inc/wallet.git
+
+# if pulling MS runtime into another MS app
+$ git remote add upstream git@github.com:mobilestack-xyz/mobilestack-runtime.git
+```
+
+## Syncing forks
+
+### Ensure local repo is up to date
+
+On the root of the forked repository
+
+```console
+$ git fetch origin
+$ git fetch upstream
+$ git checkout main
+$ git pull origin main
+```
+
+### Create branch
+
+Create a backup branch for main in case anything goes wrong. You can also do
+this directly on github
+
+```console
+$ git checkout -b main-backup
+```
+
+Create a branch for syncing changes
+
+```console
+# rename branch as required (e.g., prefixing your user id)
+$ BRANCH_NAME=<user-id>/sync-fork
+$ git checkout -b $BRANCH_NAME
+```
+
+### Merge upstream changes and resolve conflicts
+
+```console
+$ git merge upstream/main
+```
+
+If there are any conflicts, resolve them manually and commit the merge. The
+default merge commit is named `Merge branch 'source-branch' into
+'destination-branch'`, rename this to follow conventional commit format. E.g.,
+`chore: sync with [valora/mobilestack-runtime]`
+
+If you accidentally commit with the default message, you can edit the message by
+running the below, which would present you a text editor to update the commit message
+
+```console
+$ git commit --amend
+```
+
+### Open a pull request
+
+Push the remote branch and open a pull request on github. Make sure to include
+conflicted filenames in the PR description to make it easier for reviewers.
+
+```console
+$ git push origin $BRANCH_NAME
+```
+
+### Merge pull request
+
+Once the PR is approved, merge the pull request using the `Create a merge
+commit` option:
+
+![create-a-merge-commit](./assets/create-a-merge-commit.png)
+
+This is off by default and you'd need to enable `Allow merge commits` in the repo settings:
+
+![allow-merge-commit](./assets/allow-merge-commit.png)
+
+You'll get a red error message saying the repository requires linear history and
+doesn't allow merge commits. You can ignore this message and continue with the
+merge.
+
+A successful merge would mean that at the root of the repo, you'll see the
+message that says `This branch is X commits ahead of <base-repo>`. You should
+not see `Y commits behind`.
+
+Once you've merged, you can disable `Allow merge commits` option in the settings
+so other PRs are only merged using the `Squash and merge` option.
+
+## Fixing bad merges
+
+In case something went wrong with the sync, (e.g., fork is still behind base
```

Comment:
Perhaps would be good to mention that you should only attempt this/follow these instructions if there have been no new commits to main since the sync that you're trying to fix.

Reply:
updated warning below to include this info